lib/cmap/auth/gssapi.js

```diff
@@ -1,22 +1,15 @@
 'use strict';
-
-const AuthProvider = require('./auth_provider').AuthProvider;
+const { AuthProvider } = require('./auth_provider');
 const { Kerberos, kModuleError } = require('../../deps');
 
-/**
- * Creates a new GSSAPI authentication mechanism
- *
- * @class
- * @extends AuthProvider
- */
 class GSSAPI extends AuthProvider {
-  /**
-   * Implementation of authentication for a single connection
-   *
-   * @override
-   */
-  _authenticateSingleConnection(sendAuthCommand, connection, credentials, callback) {
-    const source = credentials.source;
+  auth(authContext, callback) {
+    if (Kerberos[kModuleError]) {
+      callback(Kerberos[kModuleError]);
+      return;
+    }
+
+    const { connection, credentials } = authContext;
```

Comment:
Could go a bit further with destructuring here and get rid of `credentials`, which isn't used again past here.
`const { connection, credentials: { username, password, mechanismProperties } } = authContext;`

Reply:
I thought we decided against the deep destructuring in one of Neal's PRs, because it would hose TypeScript?

Reply:
Possibly, but I don't recall that discussion. Do you remember the details of how it would hose Typescript (or perhaps you do @nbbeeken)?

Reply:
There's no issue with doing this with TS. The discussion we had is that you can't use this nested destructuring with ES6 imports, which is potentially something we'd have to use in some far off future. So I had reverted some of the requires to just use one layer of destructuring, but that didn't make it in.

Reply:
I do think it would be a bit of a downer to finally be able to use destructuring in v4, but only at the root level.